setup.py

```diff
@@ -778,14 +758,9 @@ def run(self):
             sys.exit(1)
         
         info("Testing pyzmq-%s with libzmq-%s" % (zmq.pyzmq_version(), zmq.zmq_version()))
-        
-        try:
-            import nose
-        except ImportError:
-            warn("nose unavailable, falling back on unittest. Skipped tests will appear as ERRORs.")
-            return self.run_unittest()
-        else:
-            return self.run_nose()
+        p = Popen([sys.executable, '-m', 'pytest', '-v', os.path.join('zmq', 'tests')])
+        p.wait()
+        sys.exit(p.returncode)
```

Comment:
OMG, how dusty this file is .... lots of trailing space here.

----

Hope this patch help you
See Also: https://pytest.org/latest/goodpractices.html#manual-integration
```diff
diff --git a/setup.py b/setup.py
index 3bed865..ca59aa6 100755
--- a/setup.py
+++ b/setup.py
@@ -756,52 +756,60 @@ class TestCommand(Command):
             "You must build pyzmq with 'python setup.py build_ext --inplace' for 'python setup.py test' to work.",
             "If you did build pyzmq in-place, then this is a real error."]))
             sys.exit(1)
-        
+
+        try:
+            import pytest
+        except ImportError:
+            print_exc()
+            fatal('Testing requirement "pytest" missing.')
+            sys.exit(1)
+
         info("Testing pyzmq-%s with libzmq-%s" % (zmq.pyzmq_version(), zmq.zmq_version()))
-        p = Popen([sys.executable, '-m', 'pytest', '-v', os.path.join('zmq', 'tests')])
-        p.wait()
-        sys.exit(p.returncode)
+
+        pytest_args = ['-v', os.path.join('zmq', 'tests')]
+        errno = pytest.main(pytest_args)
+        sys.exit(errno)
```